### ISSUE TITLE
Detect Solid JSX results for rendering in components.studio

### ISSUE THREAD

**Issue by georges-gomes** [ryansolid/solid#171]
I would like to enable JSX in stories

cf template:
https://components.studio/edit/sfh0anpd7QkbrE7NPNlf

Look for `index.stories.jsx`

- object: works
- jsx: works but get a warning `computations created outside a `createRoot` or `render` will never be disposed` because we can't detect it's a Solid generated Element
- fragment: can't detect yet
- dynamic component: can you help?

We discussed in gitter to possibility to watermark the base Element to detect and use `render` properly.

**Comment by ryansolid** [comment]
Object form is weird.. I don't quite get what it's doing. But since solid's render accepts functions I'm assuming why it works?

Basically all stuff needs to be executed inside a `createRoot` function. Ie... by the time you execute it's too late. Solid's render just wraps a `createRoot` call and its signature is:
```jsx
render(() => <App />, mountEl)
```
You might sometimes see:
```jsx
render(App, mountEl)
```
But that is because Solid Components are just functions and if there are no props you can just pass it as a function.

My guess is you are doing..
```jsx
render(App(), mountEl)
```
More or less. Which will never work properly unless App returns a function (ie.. the object example).
We got storybook working before like this: https://github.com/ryansolid/solid/blob/master/documentation/storybook.md

But the short of it is you have to wrap before you execute anything to create the reactive scope. So you can't determine it's a Solid Component to choose to wrap. Does that make sense?

**Comment by georges-gomes** [comment]
Yes It's not seemless but it works on JSX and Fragments now

https://components.studio/edit/sfh0anpd7QkbrE7NPNlf

But I still get `computations created outside a createRoot or render will never be disposed` which I don't understand... yet

**Comment by ryansolid** [comment]
I debugged it. It's storybook..
They literally do something like this when calling the decorator:
```
const s  = story();
decorate(() => s);
```
So frustrating.

**Comment by georges-gomes** [comment]
we don't embed storybook :)
that's our code :D

**Comment by georges-gomes** [comment]
But I can't find this code 🤔 was is the file you debug?

**Comment by georges-gomes** [comment]
I get it I think. Let me work it out

**Comment by georges-gomes** [comment]
OK fixing on our end

Could we have new method so instead of writing

``` js
export default {
  decorators: [
    (story) => {
      const d = document.createElement("div");
      render(story, d);
      return d;
    },
  ],
};
```

we could have 

```js
export default {
  decorators: [
    (story) => renderElement(story);
    },
  ],
};
```

?

To avoid the extra div and boiletplate.

What do you think?

**Comment by ryansolid** [comment]
Basically pull the div creation inside the renderElement? That's all I could do since a fragment with dynamic expressions could not return anything but an array of functions since the underlying elements could dynamically change. It is only with a parent can we have something fixed. It's reason I introduced `render`. I had been using document fragments for fragments but they have the awkward behavior of losing all their children once inserted which has a bunch of other problems.

But basically like VDOM library I need the `render` function which needs a mount element. You must be calling `render` for most libraries. If you are no longer early evaluating the only difference is wrapping with an additional function. Is the problem you don't know which render function to call in the end? Presumably somewhere down the chain it's calling `React.render` or `Preact.render` can't we call Solid`s render at the same point if we wrap it in an additional function.

**Comment by georges-gomes** [comment]
ok I see, fragment is an issue.

The stories are already functions.

```js
export const fragment = () => (
  <>
    <div>foo</div>
    <MyCounter />
    <div>bar</div>
  </>
);
```

That's why I wanted to avoid `() => () => ...`

```js
import MyCounter from "./index.js";

export const jsx = () => () => <MyCounter />;

export const fragment = () => () => (
  <>
    <div>foo</div>
    <MyCounter />
    <div>bar</div>
  </>
);
```

This works (at least on my new branch)

We detect that it's a Solid generated function and we use the right render

**Comment by georges-gomes** [comment]
It's not like anything else React or Preact (we just `() => <MyCounter/>`).
But if it's the way to go with Solid it's not a major deal breaker.

**Comment by georges-gomes** [comment]
Yes, it's better than a custom decorator with extra div.
Just needs to be documented properly.
Sounds good to you?

**Comment by ryansolid** [comment]
Sure I'm a bit used to this point the differences meaning jumping a few gaps. It's going to be a while (if ever) Solid becomes a first-class citizen, I have to carve out space against a whole ecosystem that works differently.

**Comment by wakaztahir** [comment]
@ryansolid How can I skip rendering a component if it has rendered before, meaning only once instance of element in the DOM

In SSR it doesn't work

```typescript jsx
export function CacheMountChildren(props : ParentProps<{ uniqueKey : string }>) {
    let cache = useContext(EmotionCacheContext)
        const item = cache.get(props.uniqueKey)
        if (item != null && item.hasMountedBefore) {
            // Skipped rendering
        } else {
            const setCached = (value : boolean) => cache.set(props.uniqueKey, { hasMountedBefore : value })
            setCached(true)
            onCleanup(() => setCached(false))
            return props.children
        }
 }
```

**Comment by birkskyum** [comment]
@wakaztahir, it's probably easier to reason about a new ticket.

**Comment by wakaztahir** [comment]
@birkskyum Thanks, I'll create a new ticken
